### PR TITLE
Compute running loss in tinyllama.py

### PR DESCRIPTION
**Pull request by awaelchli**
The current script logs the instantaneous loss of the last iteration in the accumulation window, it makes the loss curve a bit noisy. This PR adds a running loss over a window the size of the gradient accumulation and logs that.

Note: Instead of using torchmetrics as a dependency, we could also implement a simple RunningMean class here in lit-gpt. Let me know what you prefer.

### CODE REVIEW

File: pretrain/tinyllama.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 from lightning.fabric.utilities.throughput import ThroughputMonitor, measure_flops
 from lightning.pytorch.loggers import WandbLogger
 from torch.utils.data import DataLoader
+from torchmetrics.aggregation import RunningMean
```
Comment (awaelchli) [inline pretrain/tinyllama.py:20]: Note: Instead of using torchmetrics as a dependency, we could also implement a simple RunningMean class here in lit-gpt. Let me know what you prefer.

**Review by carmocca — APPROVED** [review]

File: pretrain/tinyllama.py (hunk 1 above)
Comment (carmocca) [reply pretrain/tinyllama.py:20]: Lit-gpt installs lightning which includes pytorch lightning which includes torchmetrics.

So you are not adding any weight here. You are fine.

Comment (awaelchli) [reply pretrain/tinyllama.py:20]: Thank you Carlos!
